### PR TITLE
Issues

### PR DESCRIPTION
**Pull request by hihihuhu**
### What problem does this PR solve?
this pr addresses a few issues:

1. add namespace to the components in operator chart, so that the manifest from `helm template` will be self contained
2. add the ability to customize tidb service selector label, which our internal service discovery system relies on
3. add the ability to customize scheduler config map annotation, so that spinnaker won't version the config map when it applied the generated manifest from `helm template`
4. ignore the new lines when set the password in tidb initializer
5. use internal service name for `discovery` in multi k8s setup, alternatively, could customize the service definition and set `ClusterIp` to `none` for discovery service as well, however, the components don't have to talk to it cross k8s cluster, thus only use internal dns record should be enough.

### What is changed and how does it work?
see section above

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
NONE
```

### CODE REVIEW

**Comment by KanShiori** [comment]
Thanks for submitting the PR.

Should we split it into multi PRs ?

**Comment by DanielZhangQD** [comment]
> Thanks for submitting the PR.
> 
> Should we split it into multi PRs ?

Agree, we should focus on one issue in one PR.
@hihihuhu Could you please help split to multiple PRs?

File: pkg/manager/member/ticdc_member_manager.go [hunk 1]
```diff
@@ -395,19 +395,18 @@ func getNewTiCDCStatefulSet(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (*ap
 		} else {
 			pdAddr = fmt.Sprintf("http://%s-pd:2379", tcName)
 		}
-		formatClusterDomain := controller.FormatClusterDomain(tc.Spec.ClusterDomain)
 
 		str := `set -uo pipefail
 pd_url="%s"
 encoded_domain_url=$(echo $pd_url | base64 | tr "\n" " " | sed "s/ //g")
-discovery_url="%s-discovery.${NAMESPACE}.svc%s:10261"
+discovery_url="%s-discovery.${NAMESPACE}.svc:10261"
```
Comment (DanielZhangQD) [inline pkg/manager/member/ticdc_member_manager.go:402]:
```suggestion
discovery_url="%s-discovery.${NAMESPACE}:10261"
```

File: pkg/manager/member/tidb_member_manager.go [hunk 2]
```diff
@@ -448,6 +448,7 @@ func getNewTiDBServiceOrNil(tc *v1alpha1.TidbCluster) *corev1.Service {
 	tidbSelector := label.New().Instance(instanceName).TiDB()
 	svcName := controller.TiDBMemberName(tcName)
 	tidbLabels := util.CombineStringMap(tidbSelector.Copy().UsedByEndUser().Labels(), svcSpec.Labels)
+	tidbSelectorLabels := util.CombineStringMap(tidbSelector.Copy().Labels(), svcSpec.SelectorLabels)
```
Comment (DanielZhangQD) [inline pkg/manager/member/tidb_member_manager.go:451]: Why do you need to update the selector? Or you actually need to update the Pod labels?

File: pkg/manager/member/tiflash_member_manager.go [hunk 3]
```diff
@@ -407,11 +407,10 @@ func getNewStatefulSet(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (*apps.St
 		} else {
 			pdAddr = fmt.Sprintf("http://%s-pd:2379", tcName)
 		}
-		formatClusterDomain := controller.FormatClusterDomain(tc.Spec.ClusterDomain)
 		str := `pd_url="%s"
 set +e
 encoded_domain_url=$(echo $pd_url | base64 | tr "\n" " " | sed "s/ //g")
-discovery_url="%s-discovery.%s.svc%s:10261"
+discovery_url="%s-discovery.%s.svc:10261"
```
Comment (DanielZhangQD) [inline pkg/manager/member/tiflash_member_manager.go:413]:
```suggestion
discovery_url="%s-discovery.%s:10261"
```

File: pkg/manager/member/template.go [hunk 4]
```diff
@@ -145,7 +145,7 @@ POD_NAME=${POD_NAME:-$HOSTNAME}
 cluster_name=` + "`" + `echo ${PEER_SERVICE_NAME} | sed 's/-pd-peer//'` + "`" +
 	`
 domain="${POD_NAME}.${PEER_SERVICE_NAME}.${NAMESPACE}.svc{{ .FormatClusterDomain }}"
-discovery_url="${cluster_name}-discovery.${NAMESPACE}.svc{{ .FormatClusterDomain }}:10261"
+discovery_url="${cluster_name}-discovery.${NAMESPACE}.svc:10261"
```
[the remark this answers is not in the document]
Comment (DanielZhangQD) [reply pkg/manager/member/template.go:148]: Please make sure that your changes will not trigger a rolling update of the current TiDB clusters deployed in one Kubernetes cluster.

**Pull request closed without merging**